src/Recorder/FilesystemRecorder.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Http\Client\Plugin\Vcr\Recorder;
+
+use GuzzleHttp\Psr7;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Symfony\Component\Filesystem\Exception\IOException;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Stores responses in a defined directory, preferably accessible to your VCS.
+ *
```

Comment:
i'd like to add an implementation note here saying that this is where we depend on guzzle psr-7 because it provides a way to convert a response to a string that can be saved and restored later.

Reply:
Is this OK for you?
```
Stores responses using the `guzzlehttp/psr7` library to serialize and deserialize the response.
Target directory should be commited.
```

Reply:
that sounds good!

Reply:
Updated as well